amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/KinesisShardDetector.java

```diff
@@ -262,4 +268,26 @@ public ListShardsResponse getListShardsResponse(ListShardsRequest request) throw
             ExecutionException, TimeoutException, InterruptedException {
         return FutureUtils.resolveOrCancelFuture(kinesisClient.listShards(request), kinesisRequestTimeout);
     }
+
+    @Override
+    public List<ChildShard> getChildShards(final String shardId) throws InterruptedException, ExecutionException, TimeoutException {
+        final GetShardIteratorRequest getShardIteratorRequest = KinesisRequestsBuilder.getShardIteratorRequestBuilder()
+                .streamName(streamIdentifier.streamName())
+                .shardIteratorType(ShardIteratorType.LATEST)
+                .shardId(shardId)
+                .build();
+
+        final GetShardIteratorResponse getShardIteratorResponse =
```

Comment:
Can we use the `getNextIterator(GetShardIteratorRequest request)` and `getGetRecordsResponse(GetRecordsRequest request)` in place of the `FutureUtils` calls?

Reply:
We are having this in ShardDetector and not DataFetcher as earlier discussed? Any concerns on implementing this for archiver?

Reply:
I think it’ll be easier to just use ShardDetector since DataFetcher will need us to override the customer’s configuration in the case of streaming readers, so it’ll avoid some overhead